content/en/docs/advanced-tasks/fail-forward-upgrades.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "Fail forward upgrades"
+linkTitle: "Fail forward upgrades"
+weight: 4
+description: >
+    Fail forward upgrades
+---
+
+## What "Fail Forward" upgrades add
+A failed installation/upgrade is typically caused by one of two scenarios:
+1. **A Failed CSV:** The CSV is in the FAILED phase.
+2. **A Failed InstallPlan:** Usually occurs because a resource listed in the InstallPlan fails to be created or updated. An InstallPlan may fail independently of its CSV and may fail to create the CSV.
+
+By opting into "failed forward" upgrades, OLM will allow you to recover from failed installations and upgrades by:
+- Allowing CSVs to move from the FAILED phase to the REPLACING phase when "fail forward" upgrades are enabled.
+- Allowing OLM to calculate new installPlans for a set of installables if:
+-- The InstallPlan referenced by a subscription is in the FAILED phase.
+-- The CatalogSource has been updated to include a new upgrade for one or more CSVs in the namespace.
+
+## Understanding "Fail Forward" upgrades
+
+Before using fail forward upgrades, it is important to understand how OLM and the Resolver calculate what needs to be installed. When determining what to install, OLM provides the resolver with the set of CSVs in a namespace. The resolver treats these CSVs differently depending on whether or not they are claimed by a Subscription, where the Subscription lists the CSV's name in its `.status.currentCSV` field.
+- If a CSV is claimed by a Subscription, the resolver will allow it to be upgraded by a bundle that replaces it.
+- If a CSV is not claimed by a Subscription, it is assumed that the user installed the CSV themselves, that OLM should not upgrade the CSV, and that it must appear in the next set of installables.
+
+OLM's Resolver is deterministic, meaning that the set of installables will not change for a given set of arguments if no new upgrades have been declared in the existing CatalogSources. When an upgrade fails due to a CSV entering the FAILED phase, the CSV being replaced still exists and is not referenced by a subscription. Today, OLM would send both `Operator v1` and `Operator v2` to the resolver which will normally be unsatisfiable because `Operator v1` is marked as required and `Operator v2` cannot be upgraded further since it provides the same APIs as `Operator v1`. In order to support "failing forward", OLM needs to omit CSVs in the REPLACING phase from the set of arguments sent to the resolver when "fail forward" upgrades are enabled.
+
+### When an InstallPlan failed
+
+Let's review what steps must be taken to recover from a failed InstallPlan today:
+1. `Operator v1` is being upgraded to `Operator v2`.
+2. The InstallPlan for `Operator v2` fails.
+3. `Operator v3` is added to the catalog and is defined as the upgrade for `Operator v1`.
+4. The user deletes the InstallPlan created for `Operator v2`.
+5. A new InstallPlan is generated for `Operator v3` and the upgrade succeeds.
```

Comment:
We may need to switch out the numerical bullet pointing with `-` to make it consistent with the rest of this markdown file. I was looking at the deploy preview and the bullet points weren't rendering there.

Reply:
Yeah, that's super weird. I tried a couple of things that other Docsy/Hugo sites have done and they weren't working. Also our entire docsite does not have a single numbered list so something must be going on there. 

Regardless, I switched these too `-`'s